vllm/model_executor/layers/quantization/utils/flashinfer_utils.py

```diff
@@ -245,14 +247,32 @@ def flashinfer_cutlass_moe_fp8(
     )
 
 
-def get_flashinfer_moe_backend() -> FlashinferMoeBackend:
+def get_flashinfer_moe_backend(layer: torch.nn.Module) -> FlashinferMoeBackend:
     flashinfer_moe_backend = envs.VLLM_FLASHINFER_MOE_BACKEND
-    if flashinfer_moe_backend == "throughput":
-        return FlashinferMoeBackend.CUTLASS
-    elif flashinfer_moe_backend == "latency":
+    from vllm.model_executor.models.llama4 import Llama4MoE
+    if flashinfer_moe_backend == "latency" or (
+            flashinfer_moe_backend == "throughput"
+            and layer.custom_routing_function
+            != Llama4MoE.custom_routing_function):
         return FlashinferMoeBackend.TENSORRT_LLM
+    elif flashinfer_moe_backend == "throughput":
```

Comment:
Does this mean that CUTLASS backend only supports Llama4's routing function?

Reply:
+1 question about if the cutlass backend only supports llama4 routing?